### DISCUSSION TITLE
secure Administrative accounts with totp?

### DISCUSSION THREAD

**Discussion by Nebdir** [kanidm/kanidm#2796] [Q&A] [answered]
Hi, i saw an issues about an removal of passkey to service accounts. We are currently enrolling yubikeys in our company and would like to secure the sensitive service accounts too. i can image that passkeys may be harder to support, but is it realistic to see one of those features in future?

**Comment by yaleman** [comment]
How would you *use* these service accounts with passkeys? They're supposed to be for automated use, and Passkeys are designed for human-in-the-loop use.

**Reply by Nebdir** [reply]
we are currently planning to supply a totp token to a request to create a user. just to verify the person has the valid permissions to do so. it would be a nice to have to bind this to a service account to enshure the request was validated by a person. otherwise we would implement this in a alternative way, like a service, dedicated to this task.

 im not shure if passkeys are really fit here, therefor i would focus on totp. am i correct?

**Reply by yaleman** [reply, marked as answer]
Service accounts are for automation, but you described a person making a create user request.

If the user has valid permissions, check their permissions, and complete the request based on that - why have a separate non-human account that the human with its own MFA to confuse the issue?

**Reply by Nebdir** [reply]
good, point. ill take this as closed, as this obviously would work against the definition.

**Comment by Firstyear** [comment]
Yeah. The intent here is service accounts use tokens that aren't manually handled by a human and can be revoked etc. 

A human on the other hand needs MFA because human handling of credentials is prone to mistakes, and MFA defends from that.

So when you think about accounts like admin/idm_admin, they are "break glass" accounts. You use them at setup to bootstrap other persons, then you enroll these persons into the relevant privilege groups like system_admins and idm_admins. 

Humans also need to re-authenticate to use their privileges (think sudo on linux or uac on windows) where they have to re-auth to access their write privileges for a short window. That makes persons safe to assign high level privs to because a token/cookie leak from the account will be in a read-only state for the majority of the time. 

Remember, MFA exists to defend credentials against *human* mistakes that we make. service-accounts can't fall for phishing emails!
